GitCommands/Git/GitCommandHelpers.cs

```diff
@@ -427,24 +427,45 @@ public static ArgumentString StopBisectCmd()
             return new GitArgumentBuilder("bisect") { "reset" };
         }
 
-        public static ArgumentString RebaseCmd(string branch, bool interactive, bool preserveMerges, bool autosquash, bool autoStash, string from = null, string onto = null)
+        public static ArgumentString RebaseCmd(string branch, bool interactive, bool preserveMerges, bool autosquash, bool autoStash, bool ignoreDate, string from = null, string onto = null)
         {
             if (from == null ^ onto == null)
             {
                 throw new ArgumentException($"For arguments \"{nameof(from)}\" and \"{nameof(onto)}\", either both must have values, or neither may.");
             }
 
-            return new GitArgumentBuilder("rebase")
+            var builder = new GitArgumentBuilder("rebase");
+            if (ignoreDate)
             {
-                { interactive, "-i" },
-                { interactive && autosquash, "--autosquash" },
-                { interactive && !autosquash, "--no-autosquash" },
-                { preserveMerges, GitVersion.Current.SupportRebaseMerges ? "--rebase-merges" : "--preserve-merges" },
-                { autoStash, "--autostash" },
-                from.QuoteNE(),
-                branch.Quote(),
-                { onto != null, $"--onto {onto}" }
-            };
+                builder.Add("--ignore-date");
+            }
+            else
+            {
+                if (interactive)
+                {
+                    builder.Add("-i");
+                    builder.Add(autosquash ? "--autosquash" : "--no-autosquash");
+                }
+
+                if (preserveMerges)
+                {
+                    builder.Add(GitVersion.Current.SupportRebaseMerges ? "--rebase-merges" : "--preserve-merges");
+                }
+            }
+
+            if (autoStash)
+            {
+                builder.Add("--autostash");
+            }
+
+            builder.AddRange(new List<string>() { from.QuoteNE(), branch.Quote() });
```

Comment:
This will cause unnecessary allocations.
For the common path I'd keep the original syntax.

Reply:
Do you mean the List declaration on line 461?

Reply:
Looking at it again, I was able to use the (boolean, string) extension method to simplify the common path. I also eliminated the useless List initialization.